discretisedfield/mesh.py

```diff
@@ -1505,3 +1544,144 @@ def axis_selector(self, *, widget="dropdown", description="axis"):
         )
+
+    def allclose(self, other, rtol=1e-05, atol=1e-08):
+        """Check if the mesh is close enough to the other based on a tolerance.
+
+        The midpoints (``discretisedfield.Mesh.points``) along all the dimensions for
+        both the mesh are compared utilizng ``numpy.allclose``. The value of relative
+        tolerance (``rtol``) and absolute tolerance (``atol``) are passed on to
+        ``numpy.allclose`` for the comparison. If the midpoints along all the
+        dimensions are close enough simultaneously, the method returns ``True``
+        otherwise it returns ``False``.
+
+        Parameters
+        ----------
+        other : discretisedfield.Mesh
+
+            The other mesh used for comparison.
+
+        rtol : int, float, optional
+
+            Relative tolerance used to compare the mesh; passed on to ``numpy.allclose``
+            . It defaults to 1e-05.
+
+        atol : int, float, optional
+
+            Absolute tolerance used to compare the mesh; passed on to ``numpy.allclose``
+            . It defaults to 1e-08.
+
+        Returns
+        -------
+        bool
+
+            ``True`` if other mesh is close enough, otherwise ``False``.
+
+        Raises
+        ------
+        TypeError
+
+            If the ``other`` argument is not of type ``discretisedfield.Mesh`` or if
+            ``rtol`` and ``atol`` arguments are not of type either ``float`` or ``int``.
+
+        ValueError
+
+            If the dimensions of the mesh and the other mesh does not match.
+
+        Example
+        -------
+        >>> p1 = (0, 0, 0)
+        >>> p2 = (20e-9, 20e-9, 20e-9)
+        >>> n = (10, 10, 10)
+        >>> mesh1 = df.Mesh(p1=p1, p2=p2, n=n)
+        ...
+        >>> p1 = (0, 0, 0)
+        >>> p2 = (20e-9 + 1.2e-12, 20e-9 + 1e-13, 20e-9 + 2e-12)
+        >>> n = (10, 10, 10)
+        >>> mesh2 = df.Mesh(p1=p1, p2=p2, n=n)
+        ...
+        >>> mesh1.allclose(mesh2, rtol=1e-6, atol=1e-11)
+        True
+        >>> mesh1.allclose(mesh2, rtol=1e-6, atol=1e-13)
+        False
+
+        """
+
+        if not isinstance(other, df.Mesh):
+            raise TypeError(
+                f"Expected argument of type discretisedfield.Mesh but got {type(other)}"
+            )
+
+        if self.region.dims != other.region.dims:
+            raise ValueError("The mesh dimensions do not match.")
+
+        if (not isinstance(rtol, (int, float))) or (not isinstance(atol, (int, float))):
+            raise TypeError(
+                "Expected both rtol and atol to be either int or float but got"
+                f" {type(rtol)} and {type(atol)} respectively"
+            )
+
+        return all(
+            [
+                np.allclose(
+                    getattr(self.points, dim),
+                    getattr(other.points, dim),
+                    rtol=rtol,
+                    atol=atol,
+                )
+                for dim in self.region.dims
+            ]
+        )
```

Comment:
The brackets inside `all(...)` can be omitted (should be done locally to run black before the commit).